### ISSUE TITLE
Feature Request: provide alternative to helm's --set-file option

### ISSUE THREAD

**Issue by xhejtman** [rancher/fleet#1711]
### Is your feature request related to a problem?

Some helm charts contains inlined files in values, such as jupyterhub chart. For pure helm install, it is possible to keep the python code in extra file and set the value using --set-file switch. It seems this is not possible with fleet. It could be defined in fleet.yaml like:
```
values:
  myvalue: "${ readFile 'somefile' }"
```
unfortunately, there is no function like readFile in templates.

Does it have any possibility? The inlined file can have hundreds of lines and it can be ugly to keep it in the values.

**Comment by weyfonk** [comment]
Would using [valuesFiles](https://fleet.rancher.io/ref-fleet-yaml#values) not be an option?

**Comment by mlhynfield** [comment]
If I may, to my understanding this request was to add a template feature to insert file contents as inline string values within the fleet.yaml.  The original example given was the `helm install --set-file` functionality, which from the `helm-install` man-pages reads:

```
--set-file=[]  set values from respective files specified via the command line (can specify multiple or separate values with commas: key1=path1,key2=path2)
```

Currently, both [valuesFiles](https://fleet.rancher.io/gitrepo-content#fleet-valuesfiles-referenced-from-fleetyaml) and [valuesFrom](https://fleet.rancher.io/gitrepo-content#using-valuesfrom) require placing the file contents (which could be python, json, really anything) inline inside of a YAML values file via block scalars, which can make editing annoying; for example:

`fleet.yaml`:

```yaml
helm:  
  valuesFiles:  
    - values.prod.yaml
```

`values.prod.yaml`:

```yaml
myValue: |
  import datetime

  def get_current_datetime():
      """Returns the current date and time as a formatted string."""
      now = datetime.datetime.now()
      return now.strftime("%Y-%m-%d %H:%M:%S")
  if __name__ == "__main__":
      print("Current Date and Time:", get_current_datetime())
```

This will certainly work, but I believe the request is for something more like this:

`fleet.yaml`:

```yaml
helm:
  values:
    myValue: "${ readFile 'datetime.py' }"
```

`datetime.py`:

```python
import datetime

def get_current_datetime():
    """Returns the current date and time as a formatted string."""
    now = datetime.datetime.now()
    return now.strftime("%Y-%m-%d %H:%M:%S")
if __name__ == "__main__":
    print("Current Date and Time:", get_current_datetime())
```

Of course this is a silly example, but there are Helm charts such as that for [Nextcloud](https://github.com/nextcloud/helm/blob/main/charts/nextcloud/values.yaml) which make use of inline files for configuration.

**Comment by xhejtman** [comment]
Hello, my point was to have structure like:
```
files/script.py
chart/...
fleet.yaml
```

and now the `fleet.yaml` can reference that `files/script.py` to be passed as `--set-file=[]`.

Main purpose is for jupyterhub helm chart where some jupyter configuration is passed via very large python file. Keeping such file in `values.yaml` is ugly (and it is what we do now). The `helm` has elegant option `--set-file=[]` to pass such file similarly to `values.yaml`, but it can be arbitrary text file, e.g., python config script. 

Basically, below is snippet from the `values.yaml` for the jupyterhub:
```yaml
hub:
  extraConfig:
    00-extra-config: |
      from traitlets import default, Unicode, List, Dict
      from tornado import gen, web
      from kubespawner import KubeSpawner
      from kubernetes_asyncio import config, client
      import asyncio
      import subprocess
      import base64
      ... 1300 more lines follow ...
```

so we would like to set `hub.extraConfig.00-extra-config=some-file.py`
